ruby_event_store/lib/ruby_event_store/specification_reader.rb

```diff
@@ -0,0 +1,35 @@
+module RubyEventStore
+  # Used for fetching events based on given query specification.
+  class SpecificationReader
+    # @api private
+    # @private
+    def initialize(repository, mapper)
+      @repository = repository
+      @mapper = mapper
+    end
+
+    # @api private
+    # @private
+    def single(specification_result)
```

Comment:
@mpraglowski if `single` isn't a term already established, I'd vote for `one`, which is the word used by ROM

Reply:
Or maybe `first` (inspired by Enumerable) - just it could be misunderstood in `Specification#last` method. `first(specification_result)` is actually what we do here.

Reply:
Another name could be `find` (also inspired by Enumerable) - the usage will be `find(specification_result)`.

Reply:
choices, choices ... ;)

Reply:
I've seen naming move closer and closer to `Enumerable`. Are you thinking that this might be the class to really align with that interface? Any reason not to do that?

Reply:
`one` appeals to me more than `single`

Reply:
So let's there be `one` ;)